spec/gilded_rose_spec.rb

```diff
@@ -3,13 +3,84 @@
 
 describe GildedRose do
 
-  describe "#update_quality" do
+  describe "item properties after running #update_quality" do
+
     it "does not change the name" do
       items = [Item.new("foo", 0, 0)]
       GildedRose.new(items).update_quality()
       expect(items[0].name).to eq "foo"
     end
-  end
 
+    it "an item's SellIn value lowers by one everyday" do
+      items = [Item.new("foo", 10, 0)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].sell_in).to eq(9)
+    end
+
+    it "a regular item's quality value decreases by 1 daily" do
+      items = [Item.new("foo", 10, 10)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq(9)
+    end
+
+    it "quality degrades twice as fast after sell by date" do
+      items = [Item.new("foo", -1, 10)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq (8)
+    end
+
+    # NEED: to deal with negative quality input in item constructor
```

Comment:
This is a really awesome and interesting edge case that I feel could come about from two scenarios: 

1) you try and create an instance of an object with a negative number
2) `update_quality` is, say, called twice times on a regular item that has a quality of 1

Those two scenarios are probably different enough that you can separate them in your testing - if we assume a 'happy path' through the application then the former won't happen, but your work here suggests that you've found an unspecified part of the current requirements. Should it throw an error? Should it create the object and just set its quality to 0?

Reply:
Cool, thanks for the input! I will deal with this after I finish the initial brief - added it to the Trello.